### PR TITLE
Add config for daily shame mail job

### PR DESCRIPTION
**Pull request by ixdy**
cc @fejta @rmmh @apelisse 

We probably should have this email our alias if it fails, though I'm not sure I want address advertised in the external repo.

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/daily-shame-mail.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                username: 'SMTP_USER'
+                password: 'SMTP_PASS'
+    builders:
+        - shell: |
+            set +x  # Don't echo credentials into the build log
+            IMAGE=gcr.io/google_containers/shame-mailer:2016-04-28-cddcf3d
+            echo "Running ${IMAGE}"
+            docker run \
+              -e GITHUB_API_TOKEN="${GITHUB_API_TOKEN}" \
+              -e SHAME_FROM="kubernetes-sig-testing+flakereport@googlegroups.com" \
```
Comment (fejta) [inline jenkins/job-configs/kubernetes-jenkins/daily-shame-mail.yaml]: reply to kubernetes-sig-testing seems a lot more useful and relevant.

**Comment by eparis** [comment]
I get why this makes sense in this repo, but I also wonder if it belongs in contrib in `mungegithub` where the actual shame mailer lives...  not sure the right answer.

File: jenkins/job-configs/kubernetes-jenkins/daily-shame-mail.yaml (hunk 1 above)
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/daily-shame-mail.yaml]: I guess I can do that. the report is sent only to Googlers, though.

Comment (eparis) [inline jenkins/job-configs/kubernetes-jenkins/daily-shame-mail.yaml]: I think you can/should add @redhat.com to the 'allowed' list as well. Although I know that isn't this PR.

Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/daily-shame-mail.yaml]: well, we can very easily cc an external email list here if desired. but if we want to start including outside folks on the test flake mail, I can change this.

Comment (eparis) [inline jenkins/job-configs/kubernetes-jenkins/daily-shame-mail.yaml]: You have permission to at a minimum update https://github.com/kubernetes/contrib/blob/master/mungegithub/reports/shame.go#L301

Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/daily-shame-mail.yaml]: Opened https://github.com/kubernetes/contrib/pull/890 to add a flag to control the allowed domains. It'd probably be good to wait on that PR, after which I'll update this one to pull in a new version of the shame mailer and add the flag allowing google.com and redhat.com.

**Comment by ixdy** [comment]
Updated this PR to use the latest shame-mailer image (just pushed) and allow sending emails to google.com and redhat.com address. PTAL.

**Comment by eparis** [comment]
I'm happy with it.

**Comment by spxtr** [comment]
LGTM

**Pull request merged**